backend/app/models.py

```diff
@@ -32,6 +36,16 @@ def __eq__(self, other):
         return self.identifier == other.identifier
 
 
+class DocumentManager(models.Manager):
+    def create_document(self, attributes):
+        doc = self.create(attributes)
+        doc.get_tokenized_text()
+        doc.get_word_count()
+        doc.get_word_count_counter()
+        doc.get_part_of_speech_tags()
+        return doc
```

Comment:
I like the direction this is going, but I see below that there's still a ton of the vestigial memoization stuff in place, which means there's a ton of boilerplate below that I'm not sure is really adding anything. Could we delete the methods `get_word_count`, `get_word_count_counter`, and `get_part_of_speech_tags` and instead do something like:
```
def create_document(self, attributes):
  text = attributes.text
  tokenized_text = Document._get_tokenized_text(text)
  word_count = len(tokenized_text)
  word_count_counter = Counter(tokenized_text)
  part_of_speech_tags = nltk.pos_tag(tokenized_text)
  document = self.create(#the above kwargs)
  return document
```

Reply:
I've moved the functionalities of `get_word_count`, `get_word_count_counter`, and `get_part_of_speech_tags` into `get_tokenized_text` and renamed it `get_tokenized_text_wc_and_pos`. This way, we'll only need to call `get_tokenized_text_wc_and_pos` when we first add in the `text` and when we update it.